packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -221,11 +221,7 @@ const hashPath = async (path: string) => {
 };
 
 const encodeFilePath = (filePath: string) => {
-	const encodedPath = filePath
-		.split(path.sep)
-		.map((segment) => encodeURIComponent(segment))
-		.join("/");
-	return "/" + encodedPath;
+	return encodeURIComponent("/" + filePath.split(path.sep).join("/"));
```

Comment:
Can this be encodeURl now since it isn't operating on each segment anymore?

```suggestion
	return encodeURI("/" + filePath.split(path.sep).join("/"));
```

Reply:
encodeURI doesn't encode characters like '/' and '#' while encodeURIComponent does